### ISSUE TITLE
Please add support for Windows Universal 10.0 and ASP.NET Core 5.0

### ISSUE THREAD

**Issue by Shekesh** [DecaTec/Portable-WebDAV-Library#1]

**Comment by DecaTec** [comment]
As the library targets Windows 8 and Windows Phone 8.1 and no special Windows 10 features are used, the library is already fully compatible with Windows 10 universal apps (UWP).

ASP .NET Core 5.0 support added https://github.com/DecaTec/Portable-WebDAV-Library/commit/fe4dd19b261e2bd679426e741f131152be16d653

**Comment by Shekesh** [comment]
When I try to use your library in an UWP 10 App with the example provided by you

```
// Specify the user credentials and use it to create a WebDavSession instance.
var credentials = new WebDavCredential("UserName", "MyPassword");
var webDavSession = new WebDavSession(@"http://www.myserver/webdav/", credentials);
var items = await webDavSession.ListAsync(@"myfolder/");
```

I get this exception: 

```
Exception thrown: 'System.PlatformNotSupportedException' in System.Net.Http.dll
Message: "The value 'DecaTec.WebDav.WebDavCredential' is not supported for property 'Credentials'."
```

**Comment by DecaTec** [comment]
I did some testing and research and unfortunately there is a problem when targeting .NET Framework (no matter what version) and Windows 8/Windows Phone 8.1/UWP: Due to .NET Framework the communication has to be done with an underlying System.Net.Http.HttpClient. These are some issues when using this with a UWP app.
When it comes to UWP, Microsoft recommends using Windows.Web.Http.HttpClient instead, but this API is not available for the .NET Framework.

So in order to support all these platforms, the library has to be split up: One part targeting .NET Framework and ASP.NET Core 5.0 and the other part targeting Windows 8/Windows Phone and UWP. I think that this will take some time to be implemented.

**Comment by DecaTec** [comment]
I've updated the library to v0.2.0.0 (see releases).

I thought that splitting the library into  two parts (.NET Framework and Windows 8/Windows 8.1 and later, UWP) would be the solution to the problem. But splitting the library wouldn't be be best solution for a portable class library.

When using the library in a UWP app:
- There was a problem using a custom credentials class (WebDavCredentials) with UWP. The credentials are submitted with the class NetworkCredential now (WebDavCredential not longer supported). -> fixed in v0.2.0.0
- There is obviously a bug in the System.Net.Http.HttpClient assembly which is referenced by a UWP project  ([see MSDN forums](https://social.msdn.microsoft.com/Forums/vstudio/en-US/ff66088c-6705-40c0-b839-491b6ed6eb76/uwpwindows-10-bug-in-httpclienthttpwebrequest-when-using-network-credentials?forum=wpdevelop)). You would get a NotSupportedException (IRandomAccessStream does not support the GetInputStreamAt method because it requires cloning and this stream does not support cloning) when trying to access WebDAV resources.
- MS is aware of this bug and it will be solved in a future release of System.Net.Http.
- For now, there is a workaround: you have to add a reference to the pre-release NuGet package 'System.Net.Http':
  Right click on your project in Visual Studio > Manage NuGet Packages... > Browse > Search for 'System.Net.Http' and tick the option Include prerelease > Now install the latest beta version of this package. 

I'll add a detailed description in the wiki of the Portable WebDAV Library.
I hope this workaround suit your needs. As soon as the System.Net.Http assembly gets updated for UWP projects, this workaround will not longer be necessary.

**Comment by DecaTec** [comment]
I'll leave this issue closed as the compatibility to ASP .NET Core 5.0 is set with v0.2.0.0 of the library.
For the issues using it in a UWP project, I've opened another issue: https://github.com/DecaTec/Portable-WebDAV-Library/issues/2
